core/include/scipp/core/element/creation.h

```diff
@@ -24,6 +24,8 @@ constexpr auto zeros_not_bool_like =
                  using T = std::decay_t<decltype(x)>;
                  if constexpr (std::is_same_v<T, bool>)
                    return int64_t{0};
+                 else if constexpr (std::is_same_v<Eigen::Vector3d, T>)
```

Comment:
Might have been safer to try to detect an Eigen templated base class and apply T::zero on that.

pseudo 
```
template<typename ...T>
struct is_eigen_matrix : std::false_type {
};

template <typename ...T>
struct is_eigen_matrix<Eigen::Matrix<T...>> : std::true_type {
};
```

Reply:
@Matthew-Andrew in fact, I think exactly this is being done in `data_model.h`